ethereumj-core/src/main/java/org/ethereum/crypto/ECKey.java

```diff
@@ -597,14 +885,18 @@ public static ECKey signatureToKey(byte[] messageHash, String signatureBase64) t
      */
     public byte[] decryptAES(byte[] cipher){
 
-        if (priv == null)
+        if (privKey == null) {
             throw new MissingPrivateKeyException();
+        }
+        if (!(privKey instanceof BCECPrivateKey)) {
+            throw new UnsupportedOperationException("Cannot use the private key as an AES key");
+        }
```

Comment:
Let's just mark it deprecated, not sure if there no projects using it for some reason